packages/util/src/array.ts

```diff
@@ -61,3 +61,26 @@ export function compare(before: any[], after: any[]): IDifferences {
     }
     return retVal;
 }
+
+export function compare2<T>(before: readonly T[], after: readonly T[], idFunc: (itme: T) => string | number): IDifferences<T> {
+    const retVal: IDifferences<T> = {
+        unchanged: [],
+        removed: [],
+        added: []
+    };
+    const addedMap: { [key: string]: T } = {};
+    after.forEach(item => {
+        addedMap[idFunc(item)] = item;
+        retVal.added.push(item);
+    });
+    for (const row of before) {
+        const otherIdx = retVal.added.indexOf(addedMap[idFunc(row)]);
```

Comment:
Shouldn't this be comparing the `idFunc` hashed representations of `item` instead of comparing item directly?

Reply:
retVal.added and addedMap are both initialized with the same objects at the start, so a straight `indexOf` should be fine.

Reply:
Nevermind, I see the purpose of compare2 now.